scripts/git-hooks/check_cli_doc.sh

```diff
@@ -0,0 +1,67 @@
+#!/bin/sh
+
+top_level=`git rev-parse --show-toplevel`
+str_functions_rst="src/doc/cli_manual/functions.rst"
+str_method_doc_C="src/visitpy/common/MethodDoc.C"
+str_method_doc_h="src/visitpy/common/MethodDoc.h"
+method_doc_staged=false
+
+# Look to see if functions.rst is staged. If it is, then run function_to_method_doc and look for
+# changes in MethodDoc.h/C. Enforce coupling between staged changes to functions.rst and resulting
+# changes in MethodDoc.h/C
+for staged in $(git diff --cached --name-only)
+do
+  if [ $staged == $str_functions_rst ]; then
+    method_doc_staged=true
+    $top_level/src/doc/functions_to_method_doc.py $top_level
+      
+    mdc_staged=false
+    mdh_staged=false
+    for unstaged in $(git diff --name-only)
+    do
+      # functions.rst should not have staged and unstaged changes because the
+      # MethodDoc.h/C will not correlate with the commiting functions.rst
+      if [ $unstaged == $str_functions_rst ]; then
+        echo "ERROR: Cannot have both staged and unstaged changes to functions.rst."
+        echo "Please stage all changes to functions.rst or stash unstaged changes."
+        exit 1
+      elif [ $unstaged == $str_method_doc_C ]; then
+        mdc_staged=true
+      elif [ $unstaged == $str_method_doc_h ]; then
+        mdh_staged=true
+      fi
+    done
+    
+    # Output error messages and exit
+    if [ $mdc_staged == true ] && [ $mdh_staged == true ]; then
+      echo "ERROR: Your changes to functions.rst will modify MethodDoc.h and MethodDoc.C, but you are not staging those."
```

Comment:
So, what do the terms *staged* and *unstaged* mean? Is this git terminology? Is it whether the changes have been *comitted*? If so, probably a little clearer to use *committed* and *uncommitted* in the interface to users (e.g. the error messages).

Reply:
I guess its whether the changed files have been *added* via `git add ...`. In which case, I think its better to just use that terminology as well as indicate the command they need to run to achieve that in the error message itself as in `Please add (or re-add) functions.rst to the commit`

Reply:
Staged is a very common term in the git vernacular that I believe is well understood. I've actually seen more confusion with the terminology "add your files". For example, "Could you please add your files to the repo" really means "make a commit". We could be explicit and say "Please add your files to the commit", but that is really the same as just saying "Please stage your files".